lib/engine/game/g_18_royal_gorge/step/buy_train.rb

```diff
@@ -16,6 +16,21 @@ def room?(entity)
             # even when train tight, there's room for a self-rust
             return true if entity.trains.any? { |t| t.rusts_on == upcoming_train.name }
           end
+
+          def check_connected!(buyer, train)
+            return if train.owner == @game.depot
```

Comment:
Adding a `return if @game.loading` guard cause here would remove the need for the game graph to be recalculated when a game is being loaded, at the cost of not rechecking the legality of the move.